pkg/controller/deployment/deployment_controller.go

```diff
@@ -356,11 +356,21 @@ func (dc *DeploymentController) deletePod(obj interface{}) {
 	glog.V(4).Infof("Pod %s deleted.", pod.Name)
 	if d := dc.getDeploymentForPod(pod); d != nil && d.Spec.Strategy.Type == extensions.RecreateDeploymentStrategyType {
 		// Sync if this Deployment now has no more Pods.
-		rsList, err := dc.getReplicaSetsForDeployment(d)
+		selector, err := metav1.LabelSelectorAsSelector(d.Spec.Selector)
 		if err != nil {
 			return
 		}
-		podMap, err := dc.getPodMapForDeployment(d, rsList)
+		rsList, err := dc.rsLister.ReplicaSets(d.Namespace).List(selector)
```

Comment:
Updated to use ListReplicaSets, had to expose a couple functions and a type in the deployment util package